test/utils.js

```diff
@@ -40,11 +40,15 @@ module.exports.sourceAndRedactedToRestored = (
   } else {
     restorationMethods = plugin.restorationMethods;
   }
-  const redactedSourceTree = unified()
+  const indexAgnosticRedactedSourceTree = unified()
```

Comment:
Minor nitpick about variable naming. I would like to seem them reflect more direct the functions which modified them.
```
  const indexAgnosticRedactedSourceTree = unified()
    .use(markdown, {commonmark: true})
    .use(redact)
    .use(plugin)
    .parse(source);
  const redactedSourceTree = unified()
    .use(markdown, {commonmark: true})
    .use(redact)
    .runSync(indexAgnosticRedactedSourceTree);
  const restorationTree = unified()
    .use(markdown, {commomark: true})
    .use(parseRestorations)
    .parse(redacted);
  const mergedTree = unified()
    .use(restore, redactedSourceTree, restorationMethods)
    .runSync(restorationTree);
```
becomes
```
  const parsedRedactedSourceTree = unified()
    .use(markdown, {commonmark: true})
    .use(redact)
    .use(plugin)
    .parse(source);
  const transformedRedactedSourceTree = unified()
    .use(markdown, {commonmark: true})
    .use(redact)
    .runSync(parsedRedactedSourceTree);
  const parsedRestorationTree = unified()
    .use(markdown, {commomark: true})
    .use(parseRestorations)
    .parse(redacted);
  const transformedRestorationTree = unified()
    .use(restore, transformedRedactedSourceTree, restorationMethods)
    .runSync(parsedRestorationTree);
 ```
 
 Or something like that. The names I gave in the above examples might be a little wordy. I'll leave it to you to come up with something clearer.